### ISSUE TITLE
Can't get "Add Scene Item (slow)" to work 

### ISSUE THREAD

**Issue by nvh86** [Zylann/godot_voxel#323]
Hi, 

Ik was playing arround with the voxel tools cool stuff b.t.w.
 
So I made a voxelLODterrain with a VoxelStreamRegionFiles Stream, a VoxelGeneratorNoise2D and the VoxmesherTransvoxel. 
Got a working terrain object. 

I added a Voxelinstancer node as a child of the VoxelLodTerain node. 
My goal was to spread trees over the terain with working collision on the character 

In the VoxelInstanceLibrary i added a scene object via the menu (Add Scene Item (slow))
I tried different hierarchies for the scene object 
- Spatial
  - Static Body
     - Mesh Instance
     - CollisionShape

- StaticBody
  - Mesh Instance 
  - Collision Shapre

The trees are actually displayed in the godot editor 3d view. 
Although when running the main scene the terrain remains empty. 

When using the multimesh option trees do appear. 
Unfortunatly collisions (and transparancy) 
won't work for this mode.

Maybe i'm approaching this in the wrong way. 
Any help would be appriciated. 
?

**Comment by Zylann** [comment]
> The trees are actually displayed in the godot editor 3d view.
Although when running the main scene the terrain remains empty.

That's odd, it should have worked in game too if you could see them in editor. Although you do need to get close enough to the LOD level they are set to spawn from.

> When using the multimesh option trees do appear.
Unfortunatly collisions (and transparancy)
won't work for this mode.

Collisions work with this mode, they need to be setup either from script, or by using the `Update from scene` menu (where the scene has to conform to the hierarchy described here https://voxel-tools.readthedocs.io/en/latest/instancing/#setting-up-a-multimesh-item-from-a-scene)

**Comment by nvh86** [comment]
The update from scene option seems to do something because when i chose the scene my trees appeared without the transparancy. 
(Using the material override fixes the transparancy, but i'll need to re unwrap the tree so it uses one material) 

Unfortunaltly this did not fix the collisions for multimesh mode. 
When instancing the used scene object manualy on the surface of the terrain it does have collisions. 

Havent been having succes with the Directly adding the scene (slow) option 
(It might be some visibility range issue indeed). 

I'll fiddle arround some more. 
Don't know if i'll get to it today, If i figure it out (or not) i'll post either what i was doing wrong or a minimal project for your review. 

Many thanks for the quick response.

**Comment by Zylann** [comment]
> (Using the material override fixes the transparancy, but i'll need to re unwrap the tree so it uses one material)

To use multiple materials on your tree the only way with multimesh at the moment is to have these materials on the mesh directly, not on the MeshInstance material slots.

**Comment by nvh86** [comment]
Thanks for the explanation, i think the material thing will work for me with some modifications. 

I added 2 projects. 

in 1 project. 
I got the voxel terrain. 
I used (Add Scene (slow)) to add the tree scene. 
Scaled it down in the generator. 
In the preview the trees become visible (but when running the scene they don't). 
After closing and opening the project it runs into problems and the terain is empty. 
Removing the item from the instancelibrary shuts down godot. (So i end up adding a new instancelibrary when i broke it). 
(Maybe i'm missing some crucial step somewhere)

In the other project (Simplescene Multimesh).
I used the Add Multimesh Item. (Cubes)
I then selected the item and chose (update Multimesh from scene) (chose the tree scene). 
The cubes turned into trees. 
I don't seem to get collision to work this way. 
(A single scene instance somewhere between the trees has working collisions)

for the sake of the post i left the materials from the scene. 
If you could give me some pointers on the collisions that would be awesome.  

[SimpleScene.zip](https://github.com/Zylann/godot_voxel/files/7449445/SimpleScene.zip)
[SimpleScene multimesh.zip](https://github.com/Zylann/godot_voxel/files/7449446/SimpleScene.multimesh.zip)

**Comment by Zylann** [comment]
SimpleScene:

Like I often do with test projects, I delete the `.import` folder to let my development version of Godot re-import the project. Turns out this project is full of errors, for example:
```
Loading resource: res://bark10.png
ERROR: Resource file not found: res://bark10.png.
   at: (core\io\resource_loader.cpp:275)
Loading resource: res://bark10_n.png
ERROR: Resource file not found: res://bark10_n.png.
   at: (core\io\resource_loader.cpp:275)
Loading resource: res://bark10_s.png
ERROR: Resource file not found: res://bark10_s.png.
   at: (core\io\resource_loader.cpp:275)
Loading resource: res://bark10.png
ERROR: Resource file not found: res://bark10.png.
   at: (core\io\resource_loader.cpp:275)
WARNING: Couldn't load external resource: res://bark10.png
     at: ResourceInteractiveLoaderText::_parse_ext_resource (scene\resources\resource_format_text.cpp:157)
Loading resource: res://bark10_s.png
ERROR: Resource file not found: res://bark10_s.png.
   at: (core\io\resource_loader.cpp:275)
WARNING: Couldn't load external resource: res://bark10_s.png
```

After the project finished importing, I see the tree scene has no textures, likely because all the errors.

I tried opening `Main.tscn`. The following errors starts spamming:
```
ERROR: Condition "scene_item->get_scene().is_null()" is true.
   at: VoxelInstancer::update_block_from_transforms (modules\voxel\terrain\instancing\voxel_instancer.cpp:867)
```
Because your `VoxelInstanceLibrary` contains a scene item with no scene assigned to it. 
After having a closer look, it is caused by the absence of a `scene` property in the item 🤦 so that means it may have worked in editor right after setting it up, but the resource cannot remember the scene it was assigned, so every subsequent reload (of the scene in editor, or launching the game) ends up with a null item. I need to fix that.

----------

Simple Scene Multimesh:

Similar errors on import, and trees also have no materials.

For the instance library item, turns out it is the same problem. Collider information is not persistent, the scene does not hold information about them so they are forgotten when the scene is closed. That needs fixing too.

**Comment by Zylann** [comment]
I also find that recent Godot 3.4 changes completely broke the `Delete` and `Update From Scene` buttons, if the library is being edited in a sub-inspector. It has become impossible to select any of the items.

**Comment by Zylann** [comment]
I fixed the saving in 536d1f11da51e01bd248be9a48b600b1459a8508 and d32dfc9c671c31bae50e08f759eb423739450d44, you may be able to try again with the latest version.

**Comment by nvh86** [comment]
The collision parts are working now in multimesh mode. 
Also when directly selecting a "scene item(slow)" it now appears in the running scene. and remains after the file closes.

Thanks for the quick fixes. :):)
